File: cs50x/session2020/problemSet6/cash/cash.py
###
#-------------------------------------------------------------------------------
# cash.py
#-------------------------------------------------------------------------------
#
# Author:       Alwin Tareen
# Created:      Nov 24, 2020
# Execution:    python3 cash.py
# Submit50:     submit50 cs50/problems/2020/x/sentimental/cash
#
# This program calculates the minimum number of coins required to give a user
# change.
#
##

# from cs50 import get_float

total = float(input("Change owed: "))
# total = get_float("Change owed: ")

while total < 0.0:
    total = float(input("Change owed: "))
    # total = get_float("Change owed: ")

total = int(total*100)
coins = 0

while total > 0:
    if total//25 > 0:
        total -= 25
    elif total//10 > 0:
        total -= 10
    elif total//5 > 0:
        total -= 5
    elif total//1 > 0:
        total -= 1
    coins += 1

print(coins) 

File: cs50x/session2020/problemSet1/cash/cash.c
/**
|-------------------------------------------------------------------------------
| cash.c
|-------------------------------------------------------------------------------
|
| Author:       Alwin Tareen
| Created:      Jun 29, 2020
| Compilation:  make cash
| Execution:    ./cash
| Check50:      check50 cs50/problems/2020/x/cash
| Submit50:     submit50 cs50/problems/2020/x/cash
|
| This program displays the minimium number of coins for a given amount.
|
*/

#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    int coins = 0;
    
    do
    {
        printf("Change owed: ");
        scanf("%f", &dollars);
    } while (dollars <= 0.0);

    int cents = round(dollars * 100);

    while (cents > 0)
    {
        if (cents/25 > 0)
            cents -= 25;
        else if (cents/10 > 0)
            cents -= 10;
        else if (cents/5 > 0)
            cents -= 5;
        else if (cents/1 > 0)
            cents -= 1;
        coins++;
    }

    printf("%d\n", coins);    

    return 0;
}

